pkg/pipeline/transform/transform_generic.go

```diff
@@ -33,21 +35,44 @@ func (g *Generic) Transform(input []config.GenericMap) []config.GenericMap {
 	output := make([]config.GenericMap, 0)
 	for _, entry := range input {
 		outputEntry := make(config.GenericMap)
-		for _, transformRule := range g.Rules {
+		for _, transformRule := range g.rules {
 			log.Debugf("transformRule = %v", transformRule)
 			outputEntry[transformRule.Output] = entry[transformRule.Input]
 		}
+		if g.maintain {
+			g.MaintainEntrichment(outputEntry, entry)
+		}
 		log.Debugf("Transform.GenericMap = %v", outputEntry)
 		output = append(output, outputEntry)
 	}
 	return output
 }
 
+func (g *Generic) MaintainEntrichment(outputEntry, inputEntry config.GenericMap) {
+	log.Debugf("entering MaintainEntrichment")
+	for key, value := range inputEntry {
+		// if key appears in list of Rules (mapOfKeys), it was already handled
+		if g.mapOfKeys[key] {
+			continue
+		}
+		outputEntry[key] = value
+	}
+}
+
 // NewTransformGeneric create a new transform
 func NewTransformGeneric(params config.StageParam) (Transformer, error) {
 	log.Debugf("entering NewTransformGeneric")
+	log.Debugf("params.Transform.Generic = %v", params.Transform.Generic)
+	mapOfKeys := make(map[string]bool)
```

Comment:
@jotak Is this "over-engineering" --- don't we just need to apply the rules on top of the original record ... adding extra fields if needed ????

Reply:
I actually don't see `mapOfKeys` being used here

Reply:
You are correct. `mapOfKeys`  is no longer used in the latest code and should be removed.